### ISSUE TITLE
Better documentation for Softmax (and others)

### ISSUE THREAD

**Issue by AcrylicShrimp** [burn-rs/burn#350]
Hi, folks! I recently found this amazing work, and now I'm changing DL framework from TensorFlow to burn-rs for my project.

The great examples is guiding me, but sadly, I don't understand how [`Softmax`](https://docs.rs/burn/latest/burn/tensor/activation/fn.softmax.html) works. Yeah, I know what is this and way to implement this in theory. But I mean, what arguments should be passed and how this implementation interacts with them. For example, `dim` parameter is not explained, so only option I have is just guessing that it will control axis to be grouped.

In short, it will be great to have clear documentations.

Thank you!

**Comment by nathanielsimard** [comment]
Better documentation is important, I'll work on improving it, especialy for operations that have a `dim` argument.
